### PR TITLE
Android camera provider

### PR DESCRIPTION
**Pull request by jasonxunxu**
This is a fully working camera provider for Android. It is efficient and can achieve pretty high frame rate by receiving frames through a `SurfaceTexture` (based on [this snippet](https://groups.google.com/forum/#!msg/kivy-users/hiyAJqjhFjo/3-WLF4MJGpwJ)).

In addition, it provides an interface to obtain current frame data as a Numpy array (requiring OpenCV Python binding `cv2`), making it suitable for computer vision apps.

It fixes issue #2831.

### CODE REVIEW

**Comment by cbenhagen** [comment]
Thank you so much for this! Could you please rebase and make a PR against kivy:master?

**Comment by tito** [comment]
This is awesome! Finally someone get it right :)

Same as @cbenhagen, can you rebase your work on master first? We wont get PR merged first in stable without going to master first.

Thanks :)

**Comment by jasonxunxu** [comment]
Thanks for the suggestion, I rebased the branch against tag 1.8.0 on master, please let me know if it is good enough.

**Comment by tito** [comment]
@jasonxunxu Your PR at the top still indicate a PR against stable-1.8.x, you need to move it to master as well :)

File: kivy/core/camera/camera_android.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        with self._buflock:
+            if self._buffer is not None:
+                self._android_camera.addCallbackBuffer(self._buffer)  # add buffer back for reuse
+            self._buffer = data
+        #print self._buffer, len(self.frame_data)  # check if frame grabbing works
+
+    def _refresh_fbo(self):
+        self._fbo.clear()
+        with self._fbo:
+            #BindTexture(texture=self._camera_texture, index=1)
```
Comment (tito) [inline kivy/core/camera/camera_android.py]: If there is no need to bind, how the samplerExternalOES is binded then?

**Comment by jasonxunxu** [comment]
@tito How can I change the PR to be against master without recreating one? I just rebased the branch, is there something else I should have done? Thanks!

**Comment by tito** [comment]
If you click on Edit, don't you see a way to change it at the top?
If not, close this one and resubmit against PR :)

**Comment by matham** [comment]
https://github.com/kivy/kivy/compare/master...jasonxunxu:camera-android is what we want.

**Comment by jasonxunxu** [comment]
@tito The Edit button only allows me to revise the title, so I have to close this PR and submit another one.

Regarding the binding thing, it surprised me that everything works fine without even binding the texture, I don't actually know why and still am very curious.

I will try to improve efficiency as you pointed out, thanks a lot!

**Comment by jasonxunxu** [comment]
The PR is resubmitted as #3250, sorry everyone for the inconvenience...

@tito We can carry on the discussion under the new PR.

**Pull request closed without merging**